MuvrKit/MKWatchConnectivity.swift

```diff
@@ -185,41 +231,80 @@ public final class MKConnectivity : NSObject, WCSessionDelegate {
-                var sampleStart: NSDate? = nil
-                var lastTime: NSDate? = nil
-                return recorder.accelerometerDataFromDate(from, toDate: to).flatMap { (recordedData: CMSensorDataList) -> MKSensorData? in
-                    let samples = recordedData.enumerate().flatMap { (_, e) -> [Float] in
-                        if let data = e as? CMRecordedAccelerometerData where isExpectedSample(data, lastTime: lastTime) {
-                            if sampleStart == nil { // first sample - set range start date
-                                sampleStart = data.startDate
-                            }
-                            lastTime = data.startDate
-                            return [Float(data.acceleration.x), Float(data.acceleration.y), Float(data.acceleration.z)]
-                        }
-                        return []
+                encoder.append(samples)
+                encoder.close(0)
+                NSLog("Generated \(from) - \(to) samples.")
+                return (fileUrl, to)
+            }
+            
+            // try to get the data from the recorder
+            guard let sdl = recorder.accelerometerDataFromDate(from, toDate: to) else {
+                NSLog("No data available.")
+                return nil
+            }
+            var firstSampleTime: NSDate? = nil
+            var lastSampleTime: NSDate? = nil
+            var encoder: MKSensorDataEncoder? = nil
+            // enumerate, creating output only if needed
+            sdl.enumerate().forEach { (_, e) in
+                if let data = e as? CMRecordedAccelerometerData where isInRange(data) {
+                    // keep track of the first sample time
+                    if firstSampleTime == nil { firstSampleTime = data.startDate }
+                    if encoder == nil {
+                        // encoder is needed
+                        encoder = MKSensorDataEncoder(target: MKFileSensorDataEncoderTarget(fileUrl: fileUrl), types: recordedTypes, samplesPerSecond: 50)
                     }
-                    return try! MKSensorData(types: recordedTypes, start: sampleStart!.timeIntervalSince1970, samplesPerSecond: 50, samples: samples)
+                    // keep track of the last sample's time
+                    lastSampleTime = data.startDate
+                    // append data to the encoder
+                    encoder!.append([Float(data.acceleration.x), Float(data.acceleration.y), Float(data.acceleration.z)])
+                }
+            }
+            // after the loop, check if we have anything to transmit
+            if let firstSampleTime = firstSampleTime, let lastSampleTime = lastSampleTime, let encoder = encoder {
+                let recordedDuration = lastSampleTime.timeIntervalSince1970 - firstSampleTime.timeIntervalSince1970
+                encoder.close(firstSampleTime.timeIntervalSince1970)
+                // check for minimum duration
+                if recordedDuration > 8.0 {
```

Comment:
does the watch have to care about this? 
The phone can deal with it when it append this chunk of data to the session's data